src/Components/Server/src/Endpoints/CircuitMatcherPolicy.cs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    internal class CircuitMatcherPolicy : MatcherPolicy, IEndpointSelectorPolicy
+    {
+        internal const string IdClaimType = "bcid";
```

Comment:
Maybe I'm being overly-cautious here, but might it be valuable to give this a more unique name? It's not hard to imagine that people might try using the claim name `bcid` to mean something completely different in their application.

Reply:
Claims need to be short, but I would accept `mabcid` (Microsoft AspNetCore Blazor Circuit ID) sounds good?

Reply:
Yeah, sounds much less likely to clash.